bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/IammeterHandler.java

```diff
@@ -0,0 +1,208 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link IammeterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author yang bo - Initial contribution
+ */
+
+@NonNullByDefault
+public class IammeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(IammeterHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private IammeterConfiguration config;
+
+    public IammeterHandler(Thing thing) {
+        super(thing);
+        config = getConfiguration();
+    }
+
+    private final int timeout = 5000;
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            try {
+                refresh();
+            } catch (IOException | JsonSyntaxException ex) {
+                logger.warn("refresh error {}", ex.getMessage());
+            }
+        }
+    }
+
+    private boolean bExtraChannelRemoved = false;
+
+    @Override
+    public void initialize() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        IammeterConfiguration config = this.config;
+        config = getConfiguration();
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        refresh();
+                    } catch (IOException | JsonSyntaxException ex) {
+                        logger.warn("refresh error {}", ex.getMessage());
+                    }
+                }
+            };
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, config.refreshInterval, TimeUnit.SECONDS);
```

Comment:
After removing the throws declaration from `refresh()`, this can be reduced:
```suggestion
            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshInterval, TimeUnit.SECONDS);
```

Reply:
changed.